### ISSUE TITLE
`no-fallthrough` chokes on if-else statements that always break

### ISSUE THREAD

**Issue by impinball** [eslint/eslint#2795]
**ESLint version:** 0.23.0

**Minimal test case:**
```json5
// .eslintrc
{"rules": {"no-fallthrough": 2}}
```

```js
// tmp.js
switch (foo) {
case bar:
  if (baz) {
    break;
  } else {
    break;
  }
default:
}

switch (foo) {
case bar:
  if (baz) {
    break;
  } else {
    break;
  }
case blarg:
}
```

```sh
# Command
$ eslint --reset tmp.js
```

**Expected output:** (nothing)

**Actual output:**
```
tmp.js
   2:0  error  Expected a "break" statement before "default"  no-fallthrough
  12:0  error  Expected a "break" statement before "case"     no-fallthrough

✖ 2 problems (2 errors, 0 warnings)
```

**Other details:**
It's failing to detect that the code breaks from the switch in both possible branches. Also, the same problem exists with `return` if the switch statement is in a function.

**Comment by btmills** [comment]
Right now none of our rules are doing any sort of branch tracking, though it's been brought up before: #566, #1128, and it was discussed most recently in #2762. To quote @nzakas from [#2762 (comment)](https://github.com/eslint/eslint/pull/2762#issuecomment-111736108):

> I just think doing full path analysis inside of a rule is way too much logic. There might be other rules that need to do path analysis as well, so if like to see if we can extract this functionality into something more general purpose and leave this rule as a very small amount of code.

**Comment by nzakas** [comment]
This is a known issue - we don't have an easy way to fix it and we likely won't have a solution anytime soon. I'm closing this as I suspect it will end up just being left open and untouched for a long time otherwise. If we can get generic path analysis working, we will have a bunch of rules to update anyway.
